Sources_v3/ChatClient_Tests.swift

```diff
@@ -143,6 +143,32 @@ class ChatClient_Tests: StressTestCase {
         XCTAssert(webSocket?.init_requestEncoder.connectionDetailsProviderDelegate === client)
     }
     
+    func test_webSocketClient_hasAllMandatoryMiddlewares() throws {
+        // Use in-memory store
+        // Create a new chat client, which in turn creates a webSocketClient
+        _ = ChatClient(
+            config: inMemoryStorageConfig,
+            workerBuilders: workerBuilders,
+            environment: testEnv.environment
+        )
+        
+        // Assert that mandatory middlewares exists
+        let middlewares = try XCTUnwrap(testEnv.webSocketClient?.init_eventNotificationCenter.middlewares)
+        // We currently have 4 mandatory middlewares, update this if you add more
+        // in `Client.eventNotificationCenter` lazy variable
+        // (3 of them added in `eventNotificationCenter` init, 1 added in `WebSocketClient` init)
```

Comment:
I wouldn't assert on the count in this case. The WS-init middleware is already tested. This should ideally cover only the middlewares set up by Client